src/audio/buffer.c

```diff
@@ -165,6 +165,50 @@ int buffer_sync_shadow_dp_queue(struct comp_buffer *buffer, size_t limit)
 }
 #endif /* CONFIG_ZEPHYR_DP_SCHEDULER */
 
+struct comp_buffer *buffer_alloc_range(uint32_t preffered_size, uint32_t minimum_size,
+				       uint32_t caps, uint32_t flags, uint32_t align,
```

Comment:
sizes should be `size_t`

Reply:
@lyakh I changed size type to `size_t` in functions which I touch in this PR. A tons of code in sof still using `uint32_t` to store size.

Reply:
@softwarecki thanks! Yes, we definitely overuse those fixed-width types like `uint32_t`

Reply:
Our 32bit days are behind us, it sounds like a FEATURE for v2.11